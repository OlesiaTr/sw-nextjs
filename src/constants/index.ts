export const BASE_URL = 'https://sw-api.starnavi.io/';

export const INITIAL_PAGE = 1;

export const NODE_X = 240;
