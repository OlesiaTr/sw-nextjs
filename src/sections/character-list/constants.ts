export const CHARACTER_LIST_TEST_ID = 'character-list-test-id'
