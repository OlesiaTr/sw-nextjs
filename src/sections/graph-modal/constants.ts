export const GRAPH_MODAL_TEST_ID = 'graph-modal-test-id';
export const MODAL_OVERLAY_TEST_ID = 'modal-overlay-test-id';
