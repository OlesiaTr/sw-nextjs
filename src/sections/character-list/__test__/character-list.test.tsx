import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import CharacterList from '../character-list';
import { CHARACTER_LIST_TEST_ID } from '../constants';
import type { Character } from '@/helpers';

describe('CharacterList', () => {
  const MOCK_FUNCTION = jest.fn();
  const MOCK_LIST: Character[] = [
    {
      name: 'Luke Skywalker',
      gender: 'male',
      birth_year: '19BBY',
      height: '172',
      mass: '77',
      hair_color: 'blond',
      skin_color: 'fair',
      eye_color: 'blue',
      homeworld: 20,
      id: 1,
      created: '2014-12-10T16:16:29.192000Z',
      edited: '2014-12-20T21:17:50.325000Z',
      films: [1, 2, 3, 4, 5, 6],
      species: [1],
      starships: [48, 59, 64, 65, 74],
      url: 'https://sw-api.starnavi.io/people/1/',
      vehicles: [38],
    },
    {
      name: 'Darth Vader',
      gender: 'male',
      birth_year: '41.9BBY',
      height: '202',
      mass: '136',
      hair_color: 'none',
      skin_color: 'white',
      eye_color: 'yellow',
      homeworld: 20,
      id: 2,
      created: '2014-12-10T16:16:29.192000Z',
      edited: '2014-12-20T21:17:50.325000Z',
      films: [1, 2, 3, 4, 5, 6],
      species: [1],
      starships: [48, 59, 64, 65, 74],
      url: 'https://sw-api.starnavi.io/people/2/',
      vehicles: [38],
    },
  ];

  const renderCharacterList = () =>
    render(<CharacterList list={MOCK_LIST} onClick={MOCK_FUNCTION} />);

  it('renders the character list correctly', () => {
    const { asFragment } = renderCharacterList();

    const characterList = screen.getByTestId(CHARACTER_LIST_TEST_ID);
    expect(characterList).toBeInTheDocument();
    expect(characterList.children).toHaveLength(MOCK_LIST.length);

    MOCK_LIST.forEach(character => {
      expect(screen.getByText(character.name)).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('handles character card clicks correctly', () => {
    renderCharacterList();

    const characterCards = screen.getAllByRole('listitem');
    characterCards.forEach((card, index) => {
      fireEvent.click(card);
      expect(MOCK_FUNCTION).toHaveBeenCalledWith(MOCK_LIST[index].id);
    });

    expect(MOCK_FUNCTION).toHaveBeenCalledTimes(MOCK_LIST.length);
  });
});
