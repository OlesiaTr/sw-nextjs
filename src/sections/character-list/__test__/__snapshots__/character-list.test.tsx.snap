// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList renders the character list correctly 1`] = `
<DocumentFragment>
  <ul
    class="m-auto grid max-w-[calc(100vw-100px)] grid-cols-1 justify-items-center gap-5 lg:grid-cols-2 xl:grid-cols-3"
    data-testid="character-list-test-id"
  >
    <li
      class="flex flex-col justify-between bg-white rounded-xl p-5 cursor-pointer shadow transition hover:scale-95 hover:bg-custom-gradient w-[400px]"
    >
      <div
        class="flex justify-between mb-3"
      >
        <h3
          class="text-2xl text-accent uppercase font-black"
        >
          Luke Skywalker
        </h3>
        <p
          class="font-medium uppercase text-lg "
        >
          male
        </p>
      </div>
      <div
        class="bg-mainGray py-1 px-3 rounded-sm border-borderGray"
      >
        <div
          class="flex flex-wrap justify-between mb-3 font-medium "
        >
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Birth Year:
            </span>
             19BBY
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Height:
            </span>
             172 cm
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Mass:
            </span>
             77 kg
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Hair Color:
            </span>
             blond
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Skin Color:
            </span>
             fair
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Eye Color:
            </span>
             blue
          </p>
        </div>
        <div>
          <p
            class="text-2xl font-bold"
          >
            Homeworld: 20
          </p>
        </div>
      </div>
    </li>
    <li
      class="flex flex-col justify-between bg-white rounded-xl p-5 cursor-pointer shadow transition hover:scale-95 hover:bg-custom-gradient w-[400px]"
    >
      <div
        class="flex justify-between mb-3"
      >
        <h3
          class="text-2xl text-accent uppercase font-black"
        >
          Darth Vader
        </h3>
        <p
          class="font-medium uppercase text-lg "
        >
          male
        </p>
      </div>
      <div
        class="bg-mainGray py-1 px-3 rounded-sm border-borderGray"
      >
        <div
          class="flex flex-wrap justify-between mb-3 font-medium "
        >
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Birth Year:
            </span>
             41.9BBY
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Height:
            </span>
             202 cm
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Mass:
            </span>
             136 kg
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Hair Color:
            </span>
             none
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Skin Color:
            </span>
             white
          </p>
          <p
            class="w-1/2 mb-2"
          >
            <span
              class="font-semibold"
            >
              Eye Color:
            </span>
             yellow
          </p>
        </div>
        <div>
          <p
            class="text-2xl font-bold"
          >
            Homeworld: 20
          </p>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
